client/secrets/rsa.go

```diff
@@ -0,0 +1,111 @@
+package secrets
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+)
+
+const (
+	DefaultEncryptionBits  = 2048
+	BlockTypeRSAPublicKey  = "RSA PUBLIC KEY"
```

Comment:
Actually, it doesn't need to be exported.
```suggestion
	blockTypeRSAPublicKey  = "RSA PUBLIC KEY"
```